hawkular-inventory-parent/hawkular-inventory-service/src/main/java/org/hawkular/inventory/service/InventoryServiceIspn.java

```diff
@@ -111,7 +112,7 @@ public void addResource(Collection<Resource> resources) {
         Map<String, IspnResource> map = resources.stream()
                 .parallel()
                 .collect(Collectors.toMap(r -> r.getId(), r -> new IspnResource(r)));
-        resource.putAll(map);
+        resource.getAdvancedCache().withFlags(Flag.IGNORE_RETURN_VALUES).putAll(map);
```

Comment:
out of curiosity, is it for better performances?

Reply:
Yes, put and remove operations than don't use the returned value can get benefit for this.
To be honest, in local modes as we have today the gain is not high, but definitely will help in clustering modes.
I introduced here as sooner or later we will work on horizontal scalability.